powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -227,71 +229,73 @@ PowerAuthEciesEncryption decryptRequest(@Nonnull HttpServletRequest request, @No
-                    final String activationId = eciesEncryption.getContext().getActivationId();
-                    if (activationId == null) {
-                        logger.warn("Activation ID is required in ECIES activation scope");
-                        throw new PowerAuthEncryptionException();
-                    }
-                    encryptorParameters = getEciesDecryptorParameters(activationId, applicationKey, ephemeralPublicKey, encryptionContext.getVersion(), nonceResponse, timestampResponse);
-                    associatedData = "3.2".equals(encryptionContext.getVersion()) ? EciesUtils.deriveAssociatedData(EciesScope.ACTIVATION_SCOPE, encryptionContext.getVersion(), applicationKey, activationId) : null;
-                }
-                case APPLICATION_SCOPE -> {
-                    encryptorParameters = getEciesDecryptorParameters(null, applicationKey, ephemeralPublicKey, encryptionContext.getVersion(), nonceResponse, timestampResponse);
-                    associatedData = "3.2".equals(encryptionContext.getVersion()) ? EciesUtils.deriveAssociatedData(EciesScope.APPLICATION_SCOPE, encryptionContext.getVersion(), applicationKey, null) : null;
-                }
-                default -> {
-                    logger.warn("Unsupported ECIES scope: {}", eciesScope);
-                    throw new PowerAuthEncryptionException();
-                }
-            }
-
-            // Prepare envelope key and sharedInfo2 parameter for encryptor
-            final byte[] secretKey = Base64.getDecoder().decode(encryptorParameters.secretKey());
-            final EciesEnvelopeKey envelopeKey = new EciesEnvelopeKey(secretKey, ephemeralPublicKeyBytes);
-            final byte[] sharedInfo2 = Base64.getDecoder().decode(encryptorParameters.sharedInfo2());
-
-            final byte[] responseData = serializeResponseData(responseObject);
-            // Encrypt response using encryptor and return ECIES cryptogram
-            final EciesParameters parametersResponse = EciesParameters.builder().nonce(nonceBytesResponse).associatedData(associatedData).timestamp(timestampResponse).build();
-            final EciesEncryptor encryptor = eciesFactory.getEciesEncryptor(envelopeKey, sharedInfo2);
-            // Store ECIES encryptor
-            eciesEncryption.setEciesEncryptor(encryptor);
-
-            final EciesPayload payload = encryptor.encrypt(responseData, parametersResponse);
-            final String encryptedDataBase64 = Base64.getEncoder().encodeToString(payload.getCryptogram().getEncryptedData());
-            final String macBase64 = Base64.getEncoder().encodeToString(payload.getCryptogram().getMac());
-            return new EciesEncryptedResponse(encryptedDataBase64, macBase64, nonceResponse, timestampResponse);
-        } catch (Exception ex) {
-            logger.debug("Response encryption failed, error: " + ex.getMessage(), ex);
-            return null;
-        }
-    }
+// TODO: UNUSED
```

Comment:
Could this be removed or do you keep it for 3.2 changes?

Reply:
OK, I'll remove it now.